src/main/java/com/careerfair/q/model/db/Student.java

```diff
@@ -2,65 +2,62 @@
 
 import com.careerfair.q.service.database.deserializer.TimestampDeserializer;
 import com.careerfair.q.util.enums.Role;
-import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.google.cloud.Timestamp;
+import com.google.cloud.firestore.annotation.Exclude;
 import com.google.cloud.firestore.annotation.PropertyName;
-import lombok.Getter;
-import lombok.Setter;
 
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
 import java.util.List;
 
-@Getter
-@Setter
 public class Student {
-
     @PropertyName("student_id")
-    @JsonProperty("student_id")
+    @Exclude
     public String studentId;
 
     @PropertyName("first_name")
-    @JsonProperty("first_name")
+    @NotEmpty
     public String firstName;
 
     @PropertyName("last_name")
-    @JsonProperty("last_name")
+    @NotEmpty
     public String lastName;
 
     @PropertyName("university_id")
-    @JsonProperty("university_id")
+    @NotEmpty
     public String universityId;
 
     @PropertyName("major")
-    @JsonProperty("major")
+    @NotEmpty
     public String major;
 
     @PropertyName("role")
-    @JsonProperty("role")
+    @NotNull(message = "You must provide Role!")
     public Role role;
 
     @PropertyName("bio")
-    @JsonProperty("bio")
+    @NotNull
     public String bio;
 
     @PropertyName("email")
-    @JsonProperty("email")
+    @NotEmpty
     public String email;
 
     @PropertyName("gpa")
-    @JsonProperty("gpa")
+    @NotNull
     public Double gpa;
 
     @PropertyName("grad_date")
-    @JsonProperty("grad_date")
+    @NotNull
     @JsonDeserialize(using = TimestampDeserializer.class)
     public Timestamp graduationDate;
 
     @PropertyName("international")
-    @JsonProperty("international")
+    @NotNull
     public Boolean international;
 
     @PropertyName("employees")
-    @JsonProperty("employees")
+    @Exclude
```

Comment:
Same here?

Reply:
so that the employees and student-id field is not pushed to the database

Reply:
Why don't you just set it using object from firebase

Reply:
it will be an extra get call, no get call is being made to get the firebase student instance in current implementation

student_id is not an 'attribute' of the collection, so didnt include because of that

Reply:
okay cool